src/type/schema.js

```diff
@@ -123,7 +124,8 @@ export class GraphQLSchema {
   _subscriptionType: ?GraphQLObjectType;
   _directives: $ReadOnlyArray<GraphQLDirective>;
   _typeMap: TypeMap;
-  _implementations: ObjMap<Array<GraphQLObjectType>>;
+  _implementations: ObjMap<Array<GraphQLObjectType | GraphQLInterfaceType>>;
+  _objectImplementations: ObjMap<Array<GraphQLObjectType>>;
```

Comment:
@mike-marcacci It would be better to have a single object map:
```
_implementations: ObjectMap<{ objects: Array<GraphQLObjectType>, interfaces: Array<GraphQLInterfaceType> }>
```

Reply:
Hey @IvanGoncharov, I was implementing this just now, and I actually disagree here. Currently, `_implementations` contains a map of objects _and_ interfaces, while `_objectImplementations` only contains objects. Changing this so that interface implementations are stored separate from objects would make us have to recombine then in `getImplementations()` which would then return new arrays each time. This has the disadvantage of creating additional unnecessary objects (which probably doesn't matter much) but also breaks any object identity comparisons (`schema.getImplementations(myInterface) === schema.getImplementations(myInterface)`).